### PR TITLE
Vendor Agent dependencies

### PR DESCRIPTION
**Pull request by ph**
This PR is only the dependencies required by the Agent. We do this to simplify the Agent PR itself and to facilitate future backport to 7.x

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -3913,6 +3863,14 @@
 			"revision": "5500ca0de0dab231b02aedabac095d43a59f31d2",
 			"revisionTime": "2019-05-22T20:56:50Z"
 		},
+		{
+			"checksumSHA1": "cFcwn0Zxefithm9Q9DioRNcGbqg=",
+			"path": "github.com/hashicorp/go-multierror",
```
Comment (ruflin) [inline vendor/vendor.json]: Don't we already use an other lib for multierror? @andrewkroh

File: vendor/vendor.json [hunk 2]
```diff
@@ -5379,12 +5369,6 @@
 			"revision": "0f29369cfe4552d0e4bcddc57cc75f4d7e672a33",
 			"revisionTime": "2019-05-07T23:52:07Z"
 		},
-		{
```
Comment (ruflin) [inline vendor/vendor.json]: I wonder where these removals come from? No files were remove in the above.

**Comment by urso** [comment]
Dependencies added:

antlr4: 👍 
oklog/ulid: 👍 

urso/ecslog: Please update to recent ecslog. The one you are vendoing is older and still contains some bugs. The update will bring in urso/diag, urso/diag-ecs, and urso/sderr (functinoality has been split up). The testing with these is much improved and minor bugs and stability issues have been fixed.

hashicorp/errwrap: Do we need this? We already have github.com/pkg/errors which provides Wrap function.

hashicorp/go-multierror: Do we need this? We already have go.uber.org/multierr

Ecslog (functionality has been moved to sderr) provides functionality from multiple error packages in one package only. Plus it is compatible to the recent Go2 proposal (pkg/errors is deprecated and unmaintained), supports multiline printing via `%+v` and integrates well into ecslog.

File: vendor/vendor.json (hunk 1 above)
Comment (ph) [reply vendor/vendor.json]: @ruflin  we do, but can we remove it in a followup PRs I don't want to mess around with refactor and merging with master.

**Comment by ph** [comment]
I will update ecslog since this should not have any impact on our code but for _errwrap_ and _multierror_ I will make a followup. I don't want to do refactor and merge on master at the same time.

File: vendor/vendor.json (hunk 2 above)
Comment (ph) [reply vendor/vendor.json]: This is strange indeed. let me recheck that.

**Comment by michalpristas** [comment]
errwrap is not used so we might clean this up, but go-multierror is used and i would keep it there and address later outside of this PR. To keep migration simple.

**Comment by ph** [comment]
@urso dependecies updated and errwrap removed.

File: vendor/vendor.json [hunk 3]
```diff
@@ -4641,6 +4663,66 @@
...
+			"path": "github.com/urso/ecslog",
+			"revision": "b79160ce24a22a95b1146a13c8f135b1a621bfd6",
+			"revisionTime": "2020-02-10T12:54:49Z"
+		},
+		{
+			"checksumSHA1": "rP0Ft8ZrRrGSMSbdx7JwdgMz770=",
+			"path": "github.com/urso/ecslog/backend",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: wrong version.

File: vendor/vendor.json [hunk 4]
```diff
@@ -4641,6 +4663,66 @@
...
+		{
+			"checksumSHA1": "3L4vNSXK+lqMQHF/nyDGwwr1bnc=",
+			"path": "github.com/urso/ecslog/backend/appender",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: wrong version

File: vendor/vendor.json [hunk 5]
```diff
@@ -4641,6 +4663,66 @@
...
+		{
+			"checksumSHA1": "c5/5Qu8SMddh+VW2DqnBR3nEwMw=",
+			"path": "github.com/urso/ecslog/backend/layout",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: wrong version.

File: vendor/vendor.json [hunk 6]
```diff
@@ -4641,6 +4663,66 @@
...
+		{
+			"checksumSHA1": "DIM/Q3FG0w7Vzz+4I+Zgy3jhzYc=",
+			"path": "github.com/urso/ecslog/ctxtree",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: this package doesn't exist.

File: vendor/vendor.json [hunk 7]
```diff
@@ -4641,6 +4663,66 @@
...
+		{
+			"checksumSHA1": "ceTMq4ANPxCFc7fOrKw9wK7SVY0=",
+			"path": "github.com/urso/ecslog/errx",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: package doesn't exist.

File: vendor/vendor.json [hunk 8]
```diff
@@ -4641,6 +4663,66 @@
...
+		{
+			"checksumSHA1": "2uz1b+A13xbrb9zKz5VW9YOST/Y=",
+			"path": "github.com/urso/ecslog/fld",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: package doesn't exist

File: vendor/vendor.json [hunk 9]
```diff
@@ -4641,6 +4663,66 @@
...
+		{
+			"checksumSHA1": "VmPq1COo1UGII2Tl+NhGR/GfFRE=",
+			"path": "github.com/urso/ecslog/fld/ecs",
+			"revision": "cb6164201e76e9e9450a36a2f9d30865da7d72ef",
+			"revisionTime": "2019-12-16T18:45:53Z"
+		},
```
Comment (urso) [inline vendor/vendor.json]: package doesn't exist

**Comment by urso** [comment]
Looks like govendor did mess with you :)

**Comment by ph** [comment]
@urso You have no idea how I hate that tool.

**Comment by urso** [comment]
> @urso You have no idea how I hate that tool.

For me the tool just hangs for some projects I want to update. tried on windows/linux/mac, with new gopath... So I sometimes resort to update the json by hand. No idea who is more lucky :P

**Comment by ph** [comment]
> For me the tool just hangs for some projects I want to update. tried on windows/linux/mac, with new gopath... So I sometimes resort to update the json by hand. No idea who is more lucky :P

Both of us are not, I am not even sure if god mod will fix all theses problems.

Anyway, I've fixed the dependencies related to ecs.

**Comment by ph** [comment]
merging failure aren't related to this change.

**Pull request merged**